Brokerages/Bitfinex/BitfinexBrokerage.cs

```diff
@@ -0,0 +1,469 @@
+
+            try
+            {
+                var response = RestClient.GetActiveOrders();
+                if (response != null)
+                {
+                    foreach (var item in response)
+                    {
+                        list.Add(new BitfinexOrder
+                        {
+                            Quantity = Convert.ToInt32(decimal.Parse(item.OriginalAmount) * divisor),
+                            BrokerId = new List<string> { item.Id.ToString() },
+                            Symbol = symbol,
+                            Time = Time.UnixTimeStampToDateTime(double.Parse(item.Timestamp)),
+                            Price = decimal.Parse(item.Price) / divisor,
+                            Status = MapOrderStatus(item),
+                            OriginalAmount = decimal.Parse(item.OriginalAmount) * divisor,
+                            RemainingAmount = decimal.Parse(item.RemainingAmount) * divisor,
+                            ExecutedAmount = decimal.Parse(item.ExecutedAmount) * divisor
+                        });
+                    }
+                }
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+            return list;
+        }
+
+        /// <summary>
+        /// Retreive orders from exchange
+        /// </summary>
+        /// <returns></returns>
+        public override List<Order> GetOpenOrders()
+        {
+
+            var list = this.GetOpenBitfinexOrders().Select(o => (Order)o).ToList();
+
+            foreach (var item in list)
+            {
+                if (item.Status != OrderStatus.Canceled && item.Status != OrderStatus.Filled && item.Status != OrderStatus.Invalid)
+                {
+                    var cached = this.CachedOrderIDs.Where(c => c.Value.BrokerId.Contains(item.BrokerId.First()));
+                    if (cached.Count() > 0 && cached.First().Value != null)
+                    {
+                        this.CachedOrderIDs[cached.First().Key] = item;
+                    }
+                }
+            }
+            return list;
+        }
+
+        /// <summary>
+        /// Retreive holdings from exchange
+        /// </summary>
+        /// <returns></returns>
+        public override List<Holding> GetAccountHoldings()
+        {
+
+            var list = new List<Holding>();
+
+            var response = RestClient.GetActivePositions();
+            foreach (var item in response)
+            {
+                var ticker = RestClient.GetPublicTicker(TradingApi.ModelObjects.BtcInfo.PairTypeEnum.btcusd, TradingApi.ModelObjects.BtcInfo.BitfinexUnauthenicatedCallsEnum.pubticker);
+                list.Add(new Holding
+                {
+                    Symbol = Symbol.Create(item.Symbol, SecurityType.Forex, Market.Bitfinex),
+                    Quantity = decimal.Parse(item.Amount) * divisor,
+                    Type = SecurityType.Forex,
+                    CurrencySymbol = "B",
```

Comment:
There was recently a change to unicode to add the official bitcoin symbol: Ƀ (U+0243). I imagine this is currently poorly supported so "B" was a compromise. Using BTC is far less ambiguous.